habitat_sim/utils/viz_utils.py

```diff
@@ -94,3 +353,83 @@ def display_video(video_file: str, height: int = 400):
         else:
             opener = "open" if sys.platform == "darwin" else "xdg-open"
             subprocess.call([opener, video_file])
+
+
+def place_agent(sim, pos=[0.0, 0.0, 0.0], rot=np.quaternion(-1, 0, 0, 0)):
+    """    
+    Places our agent in the scene using specified position and orientation parameters
+
+    Parameters
+    ----------
+    sim : Simulator, 
+        Reference to the habitat simulator object
+    pos : List, optional
+        Location in scene to place agent. The default is [0.0,0.0,0.0].
+    rot : numpy quaternion, optional
+        Orientation of agent in scene.  The default is np.quaternion(-1, 0, 0, 0)
+
+    Returns
+    -------
+    agent's scene_node transformation matrix
+
+    """
+    agent_state = habitat_sim.AgentState()
+    agent_state.position = pos
+    agent_state.rotation = rot
+    agent = sim.initialize_agent(0, agent_state)
+    return agent.scene_node.transformation_matrix()
+
+
+def remove_all_objects(sim):
+    """
+    Removes all objects from simulation world
+
+    Parameters
+    ----------
+    sim : Simulator, 
+        Reference to the habitat simulator object
+
+    Returns
+    -------
+    None.
+
+    """
+
+    for id in sim.get_existing_object_ids():
+        sim.remove_object(id)
+
+
+def simulate(sim, dt=1.0, get_frames=True):
+    """
+    Forward simulate world physics by requested duration, 
+    returning observations if requested
+
+
+    Parameters
+    ----------
+    sim : Simulator, 
+        Reference to the habitat simulator object
+
+    dt : float, optional, default
+        Duration of simulation in seconds, Defaults to 1.0
+
+    get_frames : boolean, optional, default
+        Whether to return observations from specified sensors 
+
+    Returns
+    -------
+    observations : list
+        list of 2d numpy array of ints or floats representing observation 
+        per sensor per step.
+    """
+
+    # simulate dt seconds at 60Hz to the nearest fixed timestep
+    print("Simulating " + str(dt) + " world seconds.")
+    observations = []
+    start_time = sim.get_world_time()
+    while sim.get_world_time() < start_time + dt:
+        sim.step_physics(1.0 / 60.0)
+        if get_frames:
+            observations.append(sim.get_sensor_observations())
```

Comment:
What's the impact of actually storing all the images in a lazy list like this, I imagine it's negligible. You may want to construct an np.array of observations and place them in there so they are stored contiguously for downstream tasks.

Reply:
This is how things are currently -shrug-.  Wanted to migrate things out of the colabs before I went and refactored everything :).